packages/parser/src/ethereum/index.ts

```diff
@@ -0,0 +1,180 @@
+import { BigNumber } from 'bignumber.js'
+import { ethers } from 'ethers'
+import { Tx } from '@shapeshiftoss/blockbook'
+import { Tx as ParseTx, TxSpecific as ParseTxSpecific, Token, TransferType, Transfer } from '../types'
+import { InternalTx, Network } from './types'
+import { getSigHash } from './utils'
+import * as multiSig from './multiSig'
+import * as thor from './thor'
+import * as uniV2 from './uniV2'
+import * as zrx from './zrx'
+
+export interface TransactionParserArgs {
+  network?: Network
+  midgardUrl: string
+  rpcUrl: string
+}
+
+export class TransactionParser {
+  network: Network
+
+  private thor: thor.Parser
+  private uniV2: uniV2.Parser
+  private zrx: zrx.Parser
+
+  constructor(args: TransactionParserArgs) {
+    const provider = new ethers.providers.JsonRpcProvider(args.rpcUrl)
+
+    this.network = args.network ?? Network.Mainnet
+
+    this.thor = new thor.Parser({ network: this.network, midgardUrl: args.midgardUrl })
+    this.uniV2 = new uniV2.Parser({ network: this.network, provider })
+    this.zrx = new zrx.Parser()
+  }
+
+  // return any addresses that can be detected
+  getInternalAddress(inputData: string): string | undefined {
+    switch (getSigHash(inputData)) {
+      case this.thor.transferOutSigHash:
+        return this.thor.getInternalAddress(inputData)
+      case multiSig.SENDMULTISIG_SIG_HASH:
+        return multiSig.getInternalAddress(inputData)
+      default:
+        return
+    }
+  }
+
+  async parse(tx: Tx, address: string, internalTxs?: Array<InternalTx>): Promise<ParseTx> {
+    const sendAddress = tx.vin[0].addresses?.[0] ?? ''
+    const receiveAddress = tx.vout[0].addresses?.[0] ?? ''
+
+    let result: ParseTxSpecific | undefined
+    switch (receiveAddress) {
+      case zrx.PROXY_CONTRACT: {
+        result = this.zrx.parse(tx)
+        break
+      }
+      case this.thor.routerContract: {
+        result = await this.thor.parse(tx)
+        break
+      }
+      case uniV2.ROUTER_CONTRACT: {
+        result = await this.uniV2.parse(tx)
+        break
+      }
+    }
+
+    const pTx: ParseTx = {
+      address,
+      blockHash: tx.blockHash,
+      blockHeight: tx.blockHeight,
+      blockTime: tx.blockTime,
+      caip2: 'ETH',
+      trade: result?.trade,
+      transfers: result?.transfers ?? [],
+      txid: tx.txid,
+      value: tx.value,
+    }
+
+    if (address === sendAddress) {
+      // send amount
+      const sendValue = new BigNumber(tx.value)
+      if (!sendValue.isNaN() && sendValue.gt(0)) {
```

Comment:
```suggestion
      if (sendValue.gt(0)) {
```
`sendValue.gt(0)` is false is `isNaN() === true`

Reply:
I remember you pointing this out on a previous pr. Updating. I did not include 0 value as the parsed transaction just tracks information related to actual transfer of value. A 0 value send would imply a gas value for interacting with a contract or sending a token which we will catch.